### PR TITLE
Added ThresholdedMetrics on the experimental package

### PR DESCRIPTION
**Pull request by diegokolena**
### Please check if the PR fulfills these requirements

- [ ] Include reference to internal ticket and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Relevant tests for the changes have been added
- [ ] Relevant docs have been added / updated

### CODE REVIEW

File: examples/object_detection_2d/object_detection_2d/seed_test_run.py
Comment (munkyshi) [inline examples/object_detection_2d/object_detection_2d/seed_test_run.py:1]: Let's avoid changing the 2DOD example for this until we have a good long term solution for this.

File: kolena/_experimental/workflow/thresholded.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+    @staticmethod
+    def _data_category() -> str:
+        return "METRICS"
+
+
+@dataclass(frozen=True)
+class ThresholdedMetrics(TypedDataObject[_MetricsType], metaclass=PreventThresholdOverrideMeta):
+    """
+    Represents metrics tied to a specific threshold.
```
Comment (munkyshi) [inline kolena/_experimental/workflow/thresholded.py:43]: Does this block show up in the docs when you render things locally? You might need to update `docs/reference/workflow/evaluator.md`

File: examples/object_detection_2d/object_detection_2d/seed_test_run.py
Comment (diegokolena) [reply examples/object_detection_2d/object_detection_2d/seed_test_run.py:1]: @munkyshi Which example do you recommend to modify to illustrate the ThresholdedMetrics?

Comment (munkyshi) [reply examples/object_detection_2d/object_detection_2d/seed_test_run.py:1]: I don't think we have a great public example -- for now just an internal use case for workflow testing + ensuring that what we ship is functional from an end-to-end perspective.

File: kolena/_experimental/workflow/thresholded.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+    `ThresholdedMetrics` prohibits the use of dictionary objects as field values and guarantees that
+    the threshold values remain immutable once set. For application within a particular workflow,
+    subclassing is required to define relevant metrics fields.
+
+    Usage example:
+
+    ```python
+    from kolena.workflow import MetricsTestSample
+    from kolena.workflow import ThresholdedMetrics
```
Comment (munkyshi) [inline kolena/_experimental/workflow/thresholded.py]:
```suggestion
    from kolena._experimental.workflow import ThresholdedMetrics
```

File: kolena/_experimental/workflow/thresholded.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        return "METRICS"
+
+
+@dataclass(frozen=True)
+class ThresholdedMetrics(TypedDataObject[_MetricsType], metaclass=PreventThresholdOverrideMeta):
+    """
+    Represents metrics tied to a specific threshold.
+
+    `List[ThresholdedMetrics]` should be used as a field type within `MetricsTestSample` or
+    `MetricsTestCase` from the `kolena.workflow` module. This list is meant to hold metric values
```
Comment (munkyshi) [inline kolena/_experimental/workflow/thresholded.py]: We don't support `MetricsTestCase` anymore, right?

File: mkdocs.yml [hunk 4]
```diff
@@ -31,6 +31,7 @@ nav:
           - Nesting Test Case Metrics: advanced-usage/nesting-test-case-metrics.md
           - Uploading Activation Maps: advanced-usage/uploading-activation-maps.md
           - Setting Up Natural Language Search: advanced-usage/set-up-natural-language-search.md
+          - ThresholdedMetrics: advanced-usage/thresholded-metrics.md
```
Comment (munkyshi) [inline mkdocs.yml]: This should be under `API Reference` somewhere.

File: kolena/_experimental/workflow/thresholded.py (hunk 3 above)
Comment (diegokolena) [reply kolena/_experimental/workflow/thresholded.py]: Right. Doc updated.

File: docs/reference/workflow/thresholded-metrics.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
+# ThresholdedMetrics
```
Comment (munkyshi) [inline docs/reference/workflow/thresholded-metrics.md]: Can we have this be the full import path to make it clear that this is experimental? Displaying the full path is analogous to the other pages ([example](https://github.com/kolenaIO/kolena/blob/trunk/docs/reference/workflow/evaluator.md?plain=1#L1))

```suggestion
# `kolena._experimental.workflow.ThresholdedMetrics`
```

**Review by munkyshi — APPROVED** [review]
One final docs request, but approving otherwise. Thanks for iterating.

**Pull request merged**